### PR TITLE
Generate dependencies for template specific terraform modules based on Google Provider registry schema

### PR DESCRIPTION
**Pull request by damondouglas**
See [run-terraform-schema](https://github.com/damondouglas/DataflowTemplates/tree/tf-modules-resource-schema/cicd/cmd/run-terraform-schema)'s README for a summary of this PR.

This PR submits a command-line utility that reads from the terraform Google Provider registry schema and:
- downloads the schema in JSON format for Dataflow related resources
- generates freemarker templates for use with the maven plugin system to generate Dataflow template specific modules

Some links in cicd/cmd/run-terraform-schema/README.md will resolve when #1264 merges.

The output of this PR, listed below, is the main product of this command-line utility.
- `plugins/core-plugin/src/main/resources/terraform-classic-template.tf`
- `plugins/core-plugin/src/main/resources/terraform-flex-template.tf`

### CODE REVIEW

File: cicd/internal/terraform/filter.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+// NewFilter instantiates a Filter from a Matcher.
+func NewFilter[S Schema](matcher Matcher[S]) *Filter[S] {
+	return &Filter[S]{
+		matcher: matcher,
+	}
+}
+
+// Apply the Filter's Matcher on the data. Returns a map[string]S that satisfies the Matcher.
+func (filter *Filter[S]) Apply(data map[string]S) map[string]S {
```
Comment (riteshghorse) [inline cicd/internal/terraform/filter.go:32]:
```suggestion
func (f *Filter[S]) Apply(data map[string]S) map[string]S {
```

File: cicd/internal/terraform/filter.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	result := map[string]S{}
+	for k, v := range data {
+		if filter.matcher.Match(k, v) {
```
Comment (riteshghorse) [inline cicd/internal/terraform/filter.go:35]:
```suggestion
		if f.matcher.Match(k, v) {
```

**Review by riteshghorse — APPROVED** [review]
Mostly LGTM with minor nits

File: cicd/cmd/run-terraform-schema/describe/command.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+		for _, resource := range resources[pr] {
+			addResourceCommand(prCmd, pr, resource)
+		}
+		Command.AddCommand(prCmd)
+	}
+}
+
+func addResourceCommand(parent *cobra.Command, providerUri string, resource string) {
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf(fmt.Sprintf("%s [FILE]", resource)),
```
[the remark this answers is not in the document]
Comment (damondouglas) [reply cicd/cmd/run-terraform-schema/describe/command.go:80]: Wow I can't believe I did that :p. Thank you

**Review by riteshghorse — APPROVED** [review]

**Comment by riteshghorse** [comment]
I added the `@google.com` email to my github account now but seems like the check is not picking that up for the older commit

@an2x is there a way you can manually add me to the contributor list?

**Comment by riteshghorse** [comment]
CLA updated!

**Review by Polber — APPROVED** [review]

**Pull request merged**